Parser v2

richtja opened the pull request:
Version 2 of #10. The Only changes are: 
squash of the `Name change of pars_file method` to `Parser for new input file` and rebase of this branch on top of master.

richtja committed:
Parser for new input file

Creation of parser for parsing new format of data which contains constraints
and parameters together. Example of data is in data_file_example.txt

Signed-off-by: Jan Richter <jarichte@redhat.com>

Name change of pars_file method

Signed-off-by: Jan Richter <jarichte@redhat.com>

richtja committed:
Values and id addition to Parameter

Until now parameter was number and values were sequence from 0 to n. But with new data format, parameter has it's own name and values are strings.
Before new format, parameter name and id were same value now we need two variables for that. Also for values, we need more information than just n.

Signed-off-by: Jan Richter <jarichte@redhat.com>

richtja committed:
Delete parsing from Solver

Until now Solver was parsing constraints from string.
Now Parser do this job so we don't need it inside Solver anymore.

Signed-off-by: Jan Richter <jarichte@redhat.com>

richtja committed:
Runable example

Modification of runnable example for using new data format.

Signed-off-by Jan Richter <jarichte@redhat.com>

clebergnu commented:
@richtja looks like the test has be adjusted:

https://travis-ci.org/avocado-framework/cit-ng/jobs/509378845#L553

Maybe something like:

https://github.com/clebergnu/cit-ng/blob/review_reader/tests/test_basic.py#L9

clebergnu reviewed (changes requested):
@richtja looks like the test has be adjusted:

https://travis-ci.org/avocado-framework/cit-ng/jobs/509378845#L553

Maybe something like:

https://github.com/clebergnu/cit-ng/blob/review_reader/tests/test_basic.py#L9

richtja committed:
Basic test with parser

Until now the basic test had hardcoded data. Now when we have parser we can get these data from file.

Signed-off-by: Jan Richter <jarichte@redhat.com>

clebergnu reviewed (approved):
Tests passing, LGTM.
